### PR TITLE
adjust auth server

### PR DESCRIPTION
**Pull request by wandyezj**
- remove server dependencies on packages common and cors 
- add `Access-Control` headers directly on server
- allow building on Windows
- add comment to work around building on node 20

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -20,8 +20,9 @@ Note: the installation of Node JS add paths to your operating system's PATH vari
 
 From the root of the repo:
 
+0. On Node 20: `set NODE_OPTIONS=--openssl-legacy-provider`
```
Comment (raulclouse) [inline CONTRIBUTING.md:23]: This is true for Node 18+

File: packages/server/package.json [hunk 2]
```diff
@@ -12,8 +12,6 @@
     "citest": "npm run test"
   },
   "dependencies": {
-    "common": "0.0.1",
```
Comment (raulclouse) [inline packages/server/package.json:15]: Are these unused dependencies as a result of your change, or a separate thing?  If the latter, should do it in a separate PR.

File: packages/server/src/index.ts [hunk 3]
```diff
@@ -12,22 +9,27 @@ const app = express();
...
-    .send(SERVER_HELLO_ENDPOINT.payload);
+    .send({ message: 'Hello from Script Lab' });
 });
 
-app.options('/auth', cors());
+app.options('/auth', async (_req, res) => {
+  res
+    .header('Access-Control-Allow-Headers', 'content-type')
+    .header('Access-Control-Allow-Methods', 'GET,HEAD,PUT,PATCH,POST,DELETE')
+    .header('Access-Control-Allow-Origin', '*')
```
Comment (raulclouse) [inline packages/server/src/index.ts:29]: shouldn't this just be the handful of expected client endpoints?

File: packages/server/src/index.ts [hunk 4]
```diff
@@ -46,8 +48,8 @@ app.post('/auth', async (req, res) => {
 
   res
     .contentType('application/json')
-    .status(200)
     .header('Access-Control-Allow-Origin', '*')
```
Comment (raulclouse) [inline packages/server/src/index.ts:51]: Do we still need to set the cors header here?

**Review by raulclouse — APPROVED** [review]

File: packages/server/package.json (hunk 2 above)
Comment (wandyezj) [reply packages/server/package.json:15]: As a result of the change. I cut these out because I wasn't sure if they were causing this issue somehow.

File: packages/server/src/index.ts (hunk 3 above)
Comment (wandyezj) [reply packages/server/src/index.ts:29]: It could be more specific, the goal is to mirror what we currently do in production, we can optimize later.

File: packages/server/src/index.ts (hunk 4 above)
Comment (wandyezj) [reply packages/server/src/index.ts:51]: We do in production, and this attempts to mirror that.
